gen/field.go

```diff
@@ -157,22 +214,51 @@ func (f fieldGroupGenerator) FromWire(g Generator) error {
 						<end>
 						if err != nil {
 							return err
+							// TODO: Nest the error inside a "failed to read
+							// field X of struct Y" error.
 						}
+						<if .Required>
+							<$isSet.Rotate (printf "%sIsSet" .Name)> = true
+						<end>
+						<if $isUnion><$count>++<end>
```

Comment:
What if you receive the same field ID twice? I think that should only count as a single field, and shouldn't cause an error. (let's also add a unit test for that case)